.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  # It takes the workflow approximately 30 minutes to analyze the code base
+  # so it doesn't seem to make much sense to trigger it on every PR or commit.
+  # It runs daily at 01:00 to avoid colliding with the Coverity workflow.
+  schedule:
+    - cron: '0 1 * * *'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.workflow }}-${{ matrix.language }}-${{ github.ref }}
+      cancel-in-progress: true
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - run: sudo -E .github/workflows/unit_tests.sh SETUP
+
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```

Comment:
I forgot to pin it to SHAs. I'll push a follow-up commit to https://github.com/systemd/systemd/pull/21316

Reply:
Looking at https://github.com/github/codeql-action/issues/307, I think I need to make sure Dependabot supports their release cycle. It might take a while. I'd probably just add a comment to the commit message for now